### PR TITLE
Fix i18n

### PR DESCRIPTION
**Pull request by qhantom**
## DESCRIPTION

This PR fixes the i18n implementation in the following way. This implementation replaces the current workaround where you need to pass `userRights` in dependency array of `useEffect` hook inside `_app.tsx`.

- add route protector to only navigate to routes the user has the appropriate rights for, otherwise show an access denied message

image

- check user language on render and adjust URL accordingly
- do not allow an empty language dropdown field in profile form and user form

### TO-DO

- [x] implement and update tests
- [x] update docs
- [x] pull `main` & resolve merge conflicts
- [x] check Vercel deployment

### CODE REVIEW

File: packages/app/public/locales/de/common.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
   "actions": {
     "refresh": "Daten aktualisieren",
     "delete": "Löschen",
-    "cancel": "Abbrechen"
+    "cancel": "Abbrechen",
+    "backToHome": "zu Home zurück"
```
Comment (PhilKsr) [inline packages/app/public/locales/de/common.json:29]: Wouldn't be "Zurück zur Startseite" better?

File: packages/app/public/locales/en/common.json [hunk 2]
```diff
@@ -25,7 +25,8 @@
   "actions": {
     "refresh": "Refresh data",
     "delete": "Delete",
-    "cancel": "Cancel"
+    "cancel": "Cancel",
+    "backToHome": "back to home"
```
Comment (PhilKsr) [inline packages/app/public/locales/en/common.json:29]: Starting with a capital, right? -> "Back to home" 🅱️

File: packages/app/src/components/layouts/AuthLayout.tsx [hunk 3]
```diff
@@ -137,14 +138,16 @@ export const FOOTER_LINKS: FooterLink[] = [
...
 
 export const SEARCH_ITEMS: SearchItems[] = [
-  ...NAV_LINKS,
-  ...FOOTER_LINKS,
   {
     icon: <IconSearch />,
     label: 'profileView.title',
     to: '/profile',
     description: 'profileView.description',
+    color: 'blue',
```
Comment (PhilKsr) [inline packages/app/src/components/layouts/AuthLayout.tsx:146]: Is blue correct? Or should it be the primary variable?

File: packages/app/src/pages/_app.tsx [hunk 4]
```diff
@@ -186,12 +187,12 @@ function App({
...
-    })
-
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+    if (user.locale === 'en') {
+      router.push(`${pathname}`)
+    } else {
+      router.push(`/${user.locale}${pathname}`)
+    }
+  }, [pathname, router])
```
Comment (PhilKsr) [inline packages/app/src/pages/_app.tsx:195]: 👍

File: packages/lib/package.json [hunk 5]
```diff
@@ -118,7 +118,7 @@
     "@mantine/dates": "7.5.2",
     "@mantine/hooks": "7.5.2",
     "@mantine/modals": "7.5.2",
-    "@mantine/next": "7.5.2",
+    "@mantine/next": "6.0.21",
```
Comment (PhilKsr) [inline packages/lib/package.json:121]: Do we still need this dependency at all?

**Review by PhilKsr — CHANGES_REQUESTED** [review]
I really like the solution and it's also a wonderful refactoring! Thank you for the effort!

File: packages/app/public/locales/de/common.json (hunk 1 above)
Comment (qhantom) [reply packages/app/public/locales/de/common.json:29]: Sicherlich! (was my german alright?)

File: packages/app/public/locales/en/common.json (hunk 2 above)
Comment (qhantom) [reply packages/app/public/locales/en/common.json:29]: Righty right!

File: packages/app/src/components/layouts/AuthLayout.tsx (hunk 3 above)
Comment (qhantom) [reply packages/app/src/components/layouts/AuthLayout.tsx:146]: The `color` prop is false, it does nothing. I removed it. I also realised, that a hard-coded blue was used in line `203`. I removed that as well, the default is yellow (the matching text is highlighted with the user's search string).

File: packages/lib/package.json (hunk 5 above)
Comment (qhantom) [reply packages/lib/package.json:121]: Yes, it is used in `_document.tsx` in order to support SSR for Mantine components =).

**Comment by qhantom** [comment]
> I really like the solution and it's also a wonderful refactoring! Thank you for the effort!

Thanks man!

File: packages/lib/package.json (hunk 5 above)
Comment (PhilKsr) [reply packages/lib/package.json:121]: I see, but I think that we don't need that anymore. Look at the [Mantine usage with Next.js](https://mantine.dev/guides/next/), it's not mentioned there. Also it's not used in any template from mantine with next.

File: packages/app/public/locales/de/common.json (hunk 1 above)
Comment (PhilKsr) [reply packages/app/public/locales/de/common.json]: Superlecker!

File: packages/lib/package.json (hunk 5 above)
Comment (qhantom) [reply packages/lib/package.json:121]: Ah nice!! I didn't realise that they adjusted the SSR implementation in v7, that explains why the version is still on 6 👍 Thanks for the information ℹ️  I change it.

**Comment by qhantom** [comment]
@PhilKsr please have another look 🙂

**Review by PhilKsr — APPROVED** [review]
Great progress, thanks!

**Pull request merged**